src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +30,183 @@
 class OgComplex extends EntityReferenceAutocompleteWidget {
 
   /**
-   * The OG complex widget have a special logic on order to return the groups
-   * that user can reference to.
+   * {@inheritdoc}
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $parent = parent::formElement($items, $delta, $element, $form, $form_state);
+    // todo: fix the definition in th UI level.
```

Comment:
I'm not sure we'll need a UI for this. If somebody would want to override our class, they could inject their own service, but still use the `default:og` selection handler.